src/ui-components/datatable/datatable.component.tsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+
+import { useTranslation } from "react-i18next";
+import { Link } from "react-router-dom";
+import {
+  Button,
+  DataTable,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableToolbar,
+  TableToolbarContent
+} from "carbon-components-react";
+import { Add16 } from "@carbon/icons-react";
+import "./datatable.scss";
+
+const WidgetDataTable = ({
+  title,
+  rows,
+  headers,
+  linkTo,
+  addComponent,
+  showComponent
```

Comment:
Looking at the API used in [the example](https://github.com/openmrs/openmrs-esm-patient-chart-widgets/commit/f4cdda1b59df00588f2b59322035eea991916944#diff-7c5e3c1f85a66a10957b2ee1a79ab33f87d1bdc60dc611e5be6583c42244628aR86-R91), why did you add `addComponent` to the API, and make `showComponent` take two arguments? Seems simpler to have it be a nullary function and no separate `addComponent` parameter.

Reply:
Also I'd suggest `showAddComponent` as a name. `showComponent` leaves the question "what component?"

Unless it's intended to be used more generally than just for the `add` component?